src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -1545,6 +1550,27 @@ setup_TxQ(Config const& config)
     std::uint32_t max;
     if (set(max, "maximum_txn_in_ledger", section))
         setup.maximumTxnInLedger.emplace(max);
+
+    /* The math works as expected for any value up to and including
+       MAXINT, but put a reasonable limit on this percentage so that
+       the factor can't be configured to render escalation effectively
+       moot. (There are other ways to do that, including
+       minimum_txn_in_ledger.)
+    */
+    set(setup.normalConsensusIncreasePercent,
+        "normal_consensus_increase_percent", section);
+    setup.normalConsensusIncreasePercent = boost::algorithm::clamp(
+        setup.normalConsensusIncreasePercent, 0, 1000);
+
+    /* If this percentage is outside of the 0-100 range, the results
+       are nonsensical (uint overflows happen, so the limit grows
+       instead of shrinking). 0 is not recommended.
+    */
+    set(setup.slowConsensusDecreasePercent, "slow_consensus_decrease_percent",
+        section);
+    setup.slowConsensusDecreasePercent = boost::algorithm::clamp(
+        setup.slowConsensusDecreasePercent, 0, 100);
```

Comment:
My thinking for allowing 0 is this:
1. Presumably, consensus worked well under the prior value of `txnsExpected_` (before the increase percent, at least).
2. Consensus has not worked well with the current ledger of size `size`.
3. It's going to take the minimum of the two, so if `size` is smaller, the limit _will_ shrink, and if `txnsExpected_` is smaller, it will at least not grow.

I wouldn't _recommend_ 0, but if someone wants to experiment with it on a network, I figure it should be allowed.